### PR TITLE
feat(icons): automatic flags for deprecated icons

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

This adds automatic update of icons' deprecated flags for the Icons list story.

It also adds a new story for the `lu-icon` angular component.

-----
-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2327/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7558/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2327/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7559/ui-diff)

File: stories/qa/icon/icon.stories.html [hunk 1]
```diff
@@ -113,8 +113,8 @@ <h2>Palette</h2>
 <section class="contentSection">
 	<h2>List</h2>
 	<div *ngFor="let icon of icons">
-		<span aria-hidden="true" class="lucca-icon icon-{{icon}}"></span>
-		<code class="code">icon-{{icon}}</code>
+		<span aria-hidden="true" class="lucca-icon icon-{{icon.icon}}"></span>
```
Comment (shajz) [inline stories/qa/icon/icon.stories.html:116]: Shame that we can't destruct a *ngFor

icon icon icon icon 👯

File: stories/documentation/texts/icons/icon-basic.stories.ts [hunk 2]
```diff
@@ -12,11 +12,11 @@ import { IconComponent } from '@lucca-front/ng/icon';
 	templateUrl: './icon-basic.stories.html',
 })
 class IconStory {
-	icons: readonly string[] = IconsList;
+	icons = IconsList;
 	filter = '';
 
 	public updateIcons(filter: string) {
-		this.icons = filter ? IconsList.filter((icon: string) => icon.toLowerCase().includes(filter.toLowerCase())) : IconsList;
+		this.icons = filter ? IconsList.filter((entry) => entry.icon.toLowerCase().includes(filter.toLowerCase())) : IconsList;
```
Comment (shajz) [inline stories/documentation/texts/icons/icon-basic.stories.ts:19]:
```suggestion
		this.icons = filter ? IconsList.filter(({ icon }) => icon.toLowerCase().includes(filter.toLowerCase())) : IconsList;
```
We can do it here though 😁

File: stories/documentation/texts/icons/icon-basic.stories.html [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 <input type="checkbox" id="showLegacyIcons" class="showLegacy u-marginBottomM" />
 <label for="showLegacyIcons"> Afficher les icônes dépréciées</label>
 
 <div class="icons">
-	<div class="demo-icon" *ngFor="let icon of icons" (click)="copyIcon(icon)">
-		<span aria-hidden="true" class="lucca-icon icon-{{icon}}"></span>
-		<code class="code">icon-{{icon}}</code>
-	</div>
+  <div class="demo-icon" *ngFor="let icon of icons" (click)="copyIcon(icon.icon)">
+    <span aria-hidden="true" class="lucca-icon icon-{{icon.icon}}" [class.deprecated]="icon.deprecated"></span>
```
Comment (shajz) [inline stories/documentation/texts/icons/icon-basic.stories.html:11]: What does the `deprecated` class do ?

Comment (Supamiu) [reply stories/documentation/texts/icons/icon-basic.stories.html:11]: Hide the icon and show it if the checkbox above it checked, see `styles.scss`.

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2327/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7560/ui-diff)

**Review by shajz — DISMISSED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2327/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/7573/ui-diff)

**Review by jeremie-lucca — APPROVED** [review]

**Review by shajz — APPROVED** [review]

**Pull request merged**